homeassistant/components/intellifire/climate.py

```diff
@@ -0,0 +1,122 @@
+    """Intellifire climate entity."""
+
+    entity_description: ClimateEntityDescription
+
+    _attr_hvac_modes = [HVACMode.HEAT, HVACMode.OFF]
+    _attr_min_temp = 0
+    _attr_max_temp = 37
+    _attr_supported_features = ClimateEntityFeature.TARGET_TEMPERATURE
+    _attr_target_temperature_step = 1.0
+    _attr_temperature_unit = TEMP_CELSIUS
+
+    def __init__(
+        self,
+        coordinator: IntellifireDataUpdateCoordinator,
+        description: ClimateEntityDescription,
+    ) -> None:
+        """Configure climate entry specifically which setpoint to use for last_temp."""
+        super().__init__(coordinator, description)
+
+        if coordinator.data.thermostat_on:
+            self.last_temp = coordinator.data.thermostat_setpoint_c
+        else:
+            self.last_temp = DEFAULT_THERMOSTAT_TEMP
+
+    @property
+    def hvac_mode(self) -> str:
+        """Return current hvac mode."""
+        if self.coordinator.read_api.data.thermostat_on:
+            return HVAC_MODE_HEAT
+        return HVAC_MODE_OFF
+
+    async def async_set_temperature(self, **kwargs) -> None:
+        """Turn on thermostat by setting a target temperature."""
+        raw_target_temp = kwargs[ATTR_TEMPERATURE]
+        self.last_temp = int(raw_target_temp)
+        LOGGER.debug(
+            "Setting target temp to %sc %sf",
+            int(raw_target_temp),
+            (raw_target_temp * 9 / 5) + 32,
+        )
+        await self.coordinator.control_api.set_thermostat_c(
+            fireplace=self.coordinator.control_api.default_fireplace,
+            temp_c=self.last_temp,
+        )
+
+    @property
+    def current_temperature(self) -> float:
+        """Return the current temperature."""
+        return float(self.coordinator.read_api.data.temperature_c)
+
+    @property
+    def target_temperature(self) -> float:
+        """Return target temperature."""
+        return float(self.coordinator.read_api.data.thermostat_setpoint_c)
+
+    async def async_set_hvac_mode(self, hvac_mode: str) -> None:
+        """Set HVAC mode to normal or thermostat control."""
+        LOGGER.debug(
+            "Setting mode to [%s] - using last temp: %s", hvac_mode, self.last_temp
+        )
+
+        if hvac_mode == HVAC_MODE_HEAT:
+            # 1) Set the desired target temp
+            await self.coordinator.control_api.set_thermostat_c(
+                fireplace=self.coordinator.control_api.default_fireplace,
+                temp_c=self.last_temp,
+            )
+
+            # 2) Make sure the fireplace is on!
+            if not self.coordinator.read_api.data.is_on:
+                await self.coordinator.control_api.flame_on(
+                    fireplace=self.coordinator.control_api.default_fireplace,
+                )
+
+        elif hvac_mode == HVAC_MODE_OFF:
+            await self.coordinator.control_api.turn_off_thermostat(
+                fireplace=self.coordinator.control_api.default_fireplace
+            )
```

Comment:
You could save some indent here by moving this above, switch it to an `if` and then return inside the if block.

The you could outdent the other block since you know is heat as its the only other valid mode